app.py

```diff
@@ -1,134 +1,172 @@
-import os
-from flask import Flask, render_template, jsonify, request
+from flask import Flask, render_template, jsonify, request, Response
 import requests
-from urllib.parse import urlparse, quote_plus
+from urllib.parse import quote_plus
```

Comment:
The inclusion of type annotations enhances code clarity, improving maintainability and readability. However, ensure that these types align with actual runtime types, especially for JSON and requests where data might vary. Consider using more specific types or type aliases for complex structures, like API responses, to further clarify.